pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1463,13 +1497,55 @@ func doActivate(ctx *domainContext, config types.DomainConfig,
 			// do nothing
 		case zconfig.Format_CONTAINER:
 			snapshotID := containerd.GetSnapshotID(ds.FileLocation)
-			if err := ctx.casClient.MountSnapshot(snapshotID, cas.GetRoofFsPath(ds.FileLocation)); err != nil {
+			rootPath := cas.GetRoofFsPath(ds.FileLocation)
+			if err := ctx.casClient.MountSnapshot(snapshotID, rootPath); err != nil {
 				err := fmt.Errorf("doActivate: Failed mount snapshot: %s for %s. Error %s",
 					snapshotID, config.UUIDandVersion.UUID, err)
 				log.Error(err.Error())
 				status.SetErrorNow(err.Error())
 				return
 			}
+
+			metadataPath := filepath.Join(rootPath, "meta-data")
+
+			// get current cloud init version
+			curCIVersion, err := getVersionFromMetaFile(metadataPath)
+			if err != nil {
+				log.Error("Failed to get cloud init version from meta-data file: ", err)
```

Comment:
Are we going to log an error the first time we boot a container? I don't see how an initial meta-data file is written before we try to activate aka boot it.

Reply:
yes, is this problematic? maybe this should be a warning instead of an error, since it's not critical and we have a fallback in place

Reply:
If it is normal that the file dosn't exist it should be neither error nor warnings. We need to get rid of false errors.

Reply:
I removed the error now